PetShare/Controllers/AdopterController.cs

```diff
@@ -27,15 +30,22 @@ public AdopterController(IAdopterQuery query, IAdopterCommand command, TokenVali
     /// </summary>
     [HttpGet]
     [Authorize(Roles = Roles.Admin)]
-    [ProducesResponseType(typeof(IReadOnlyList<AdopterResponse>), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(PaginatedAdoptersResponse), StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
-    public async Task<ActionResult<IReadOnlyList<AdopterResponse>>> GetAll()
+    public async Task<ActionResult<PaginatedAdoptersResponse>> GetAll([FromQuery] PaginationQueryRequest paginationQuery)
     {
         if (!await _validator.ValidateClaims(User))
             return Unauthorized();
 
-        return (await _query.GetAllAsync()).Select(a => a.ToResponse()).ToList();
+        var adopters = (await _query.GetAllAsync()).Select(a => a.ToResponse()).ToList();
+
+        var paginatedAdopters = _paginator.GetPage<AdopterResponse>(adopters, paginationQuery);
+        if (paginatedAdopters == null)
+            return BadRequest("Wrong pagination parameters");
+
+        return PaginatedAdoptersResponse.FromPaginatedResult(paginatedAdopters.Value);
```

Comment:
Why do we need `.Value` here? `PaginatedAdoptersResponse` is a reference type.

Reply:
c# wanted this because of nullablility

Reply:
Now after changing PagiantionRelust form struct to record value is indeed unnecessary